File: app/src/main/java/net/squanchy/onboarding/OnboardingPersister.java
package net.squanchy.onboarding;

import android.content.SharedPreferences;

import java.util.Locale;

public class OnboardingPersister {

    private final SharedPreferences preferences;

    public OnboardingPersister(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    void savePageSeen(OnboardingPage page) {
        preferences.edit()
                .putBoolean(preferenceKeyFor(page), true)
                .apply();
    }

    boolean pageSeen(OnboardingPage page) {
        return preferences.getBoolean(preferenceKeyFor(page), false);
    }

    private String preferenceKeyFor(OnboardingPage page) {
        return String.format("key_onboarding_%s_seen", page.name().toLowerCase(Locale.US));
    }
}

File: app/src/main/java/net/squanchy/onboarding/OnboardingPersister.kt
package net.squanchy.onboarding

import android.content.SharedPreferences

import java.util.Locale

class OnboardingPersister(private val preferences: SharedPreferences) {

    internal fun savePageSeen(page: OnboardingPage) =
        preferences.edit()
            .putBoolean(preferenceKeyFor(page), true)
            .apply()

    internal fun pageSeen(page: OnboardingPage) = preferences.getBoolean(preferenceKeyFor(page), false)

    private fun preferenceKeyFor(page: OnboardingPage) = "key_onboarding_${page.name.toLowerCase(Locale.US)}_seen"
}
